### PR TITLE
APP-2839: add simple floating context menu component

### PR DESCRIPTION
**Pull request by mcous**
Check out the Playground to see the component in action. I don't necessarily _love_ the API, but I think this is about as far as I can take it without actually seeing it in action.

It also doesn't feel like `floating-ui` is super necessary at this stage, but it was how it was implemented in Vue so 🤷

### CODE REVIEW

**Comment by ehhong** [comment]
code looks good to me! from manually testing in the playground:
- for the floating menu example, the context menu should close after clicking "Close menu" (the control slot)
- [q] would there be any way to add minor changes to the floating position with this interface? for example, if i want the menu to float just a bit more below the control slot, can i include additive style changes?

File: packages/core/src/lib/context-menu/__tests__/floating-menu.spec.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import { writable } from 'svelte/store';
+
+import Subject from './floating-menu.spec.svelte';
+
+describe('<FloatingMenu> component', () => {
+  let user: ReturnType<typeof userEvent.setup>;
+
+  beforeEach(() => {
+    user = userEvent.setup();
+  });
```
Comment (ethanlook) [inline packages/core/src/lib/context-menu/__tests__/floating-menu.spec.ts:13]: Nice, we should broadcast this

**Review by ethanlook — APPROVED** [review]
Do you mind adding example usage to the storybook as well? Thanks!

**Comment by ehhong** [comment]
changes look great, thanks so much!! last thing -- should the context menu close when an item is clicked? or can/should that be controlled by the parent element?

**Comment by mcous** [comment]
@ehhong good catch, I exposed a quick `closeMenu` slot prop so that the parent can wire up a close action to each item's `on:click` handler. I don't think this is a great API, but it'll allow us to wire it up so we can see it all in action before coming up with something better

**Review by ehhong — APPROVED** [review]
lgtm!! thanks again for taking this on 🎊

**Pull request merged**